03week/towersOfHanoi.js

```diff
@@ -13,33 +13,81 @@ let stacks = {
   c: []
 };
 
-function printStacks() {
+const printStacks=()=> {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+const movePiece=(startStack, endStack)=> {
   // Your code here
 
-}
+  // use the push and pop method to remove the last number in a given array and move it to the last position in a different array
+  return stacks[endStack].push(stacks[startStack].pop());
+
 
-function isLegal() {
-  // Your code here
 
 }
 
-function checkForWin() {
+const isLegal = (startStack, endStack) => {
   // Your code here
+  //determin if the number your moving is being placed on a number of greater value.  If it is, return the boolean true value.  if not, return fales.
+  if (stackLetterIsLegal(startStack, endStack)) {
+  const valueOfFirstNumber = stacks[startStack][stacks[startStack].length-1]
+
+  const valueOfSecondNumber = stacks[endStack][stacks[endStack].length-1]
+
+  
+    if (valueOfFirstNumber < valueOfSecondNumber || stacks[endStack].length === 0) {
+      return true;
+   } else {
+    return false;
+   }
+  } 
+}
+// check if the move is to a valid array.  
+const stackLetterIsLegal = (startStack, endStack) => {
+  if(startStack === 'a' && endStack === 'b' || endStack === 'c') {
+    return true;
+  } else if (startStack === 'b' && endStack === 'a' || endStack === 'c') {
+    return true;
+  } else if (startStack === 'c' && endStack === 'a' || endStack === 'b') {
+    return true;
+  } else {
+    return false;
+  }
+}
 
+const checkForWin = () => {
+  // Your code here
+//if the value of the last array is 4 numerals long, then its a win.  if so, return true.  if not return false.
+  if (stacks.c.length === 4) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
-function towersOfHanoi(startStack, endStack) {
+const towersOfHanoi = (startStack, endStack) => {
   // Your code here
 
+  //put all of the above variables in order, nested within towersOfHanoi.
+  if (isLegal(startStack, endStack)) {
+  movePiece(startStack, endStack);
+  } else {
+    console.log("Not Legal Move");
+  }
+  if (checkForWin() ==true){
```

Comment:
this isn't good practice, just check for a truthy value, you don't ever want to just check for the boolean true